src/features/site-survey/actions.ts

```diff
@@ -214,6 +220,49 @@ export const updateModifiedFeatures = (
     }
   });
 
+/**
+ * Action that adjusts home positions to the current drone positions
+ * if possible.
+ */
+export const adjustHomePositionsToDronePositions =
+  (): AppThunk => async (dispatch, getState) => {
+    // Only outdoor shows are supported by the dialog.
+
+    const distanceFunction = haversineDistance;
+    const homePositions = getHomePositionsInWorldCoordinates(getState());
+    const dronePositions = getCurrentGPSPositionsOfActiveUAVs(getState());
+    if (homePositions === undefined || dronePositions.includes(undefined)) {
+      return;
```

Comment:
A snackbar message would be nice here to let the user know why nothing is happening after clicking on the button.

Reply:
The button is disabled if these checks fail (I don't think the user should ever be able to execute this action in that case), but I can still add a snackbar message. We could also show a snackbar unconditionally at the end, displaying how many home positions were updated. I'm not yet sure about these UX preferences in the project.

Reply:
If this branch is not reachable via normal user action, then a `console.warn(...)` should be enough, no need for a snackbar message. Up to you.

Reply:
I added the warning (committed directly to `dev`).